File: mobile/src/main/java/net/mm2d/dmsexplorer/viewmodel/helper/MovieActivityPipHelper.java
/*
 * Copyright (c) 2017 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.dmsexplorer.viewmodel.helper;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.View;

import net.mm2d.dmsexplorer.viewmodel.ControlPanelModel;

/**
 * @author <a href="mailto:ryo@mm2d.net">大前良介 (OHMAE Ryosuke)</a>
 */
public interface MovieActivityPipHelper {
    void register();

    void unregister();

    void setControlPanelModel(@Nullable ControlPanelModel model);

    void enterPictureInPictureMode(@NonNull View contentView);
}

File: mobile/src/main/java/net/mm2d/dmsexplorer/viewmodel/helper/MovieActivityPipHelper.kt
/*
 * Copyright (c) 2017 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.dmsexplorer.viewmodel.helper

import android.view.View
import net.mm2d.dmsexplorer.viewmodel.ControlPanelModel

/**
 * @author [大前良介 (OHMAE Ryosuke)](mailto:ryo@mm2d.net)
 */
interface MovieActivityPipHelper {
    fun register()
    fun unregister()
    fun setControlPanelModel(model: ControlPanelModel?)
    fun enterPictureInPictureMode(contentView: View)
}
